search_knn.go

```diff
@@ -142,24 +154,36 @@ func copySearchRequest(req *SearchRequest) *SearchRequest {
 		SearchAfter:      req.SearchAfter,
 		SearchBefore:     req.SearchBefore,
 		KNN:              req.KNN,
+		KNNOperator:      req.KNNOperator,
 	}
 	return &rv
 
 }
 
-func disjunctQueryWithKNN(req *SearchRequest) query.Query {
+var (
+	knnOperatorAnd = knnOperator("and")
+	knnOperatorOr  = knnOperator("or")
+)
+
+func queryWithKNN(req *SearchRequest) (query.Query, error) {
 	if len(req.KNN) > 0 {
-		disjuncts := []query.Query{req.Query}
+		subQueries := []query.Query{req.Query}
 		for _, knn := range req.KNN {
 			if knn != nil {
 				knnQuery := query.NewKNNQuery(knn.Vector)
 				knnQuery.SetFieldVal(knn.Field)
 				knnQuery.SetK(knn.K)
 				knnQuery.SetBoost(knn.Boost.Value())
-				disjuncts = append(disjuncts, knnQuery)
+				subQueries = append(subQueries, knnQuery)
 			}
 		}
-		return query.NewDisjunctionQuery(disjuncts)
+		if req.KNNOperator == knnOperatorAnd {
```

Comment:
my suggestion over here would be do some string processing like trim and lowercase conversion such that there is a little flexibility for the user to enter the operator

Reply:
Didn't add that since match query operator doesn't seem to have it either - 
https://github.com/blevesearch/bleve/blob/master/search/query/match.go#L64-L70

WDYT?

Reply:
okay cool